### PR TITLE
Fix for Agriculture

### PR DESCRIPTION
**Pull request by Skau**
Fix for #28

### CODE REVIEW

File: src/CommunityPatch/Patches/AgriculturePerkPatch.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using TaleWorlds.Core;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches {
+
+  sealed class AgriculturePatch : PatchBase<AgriculturePatch> {
+
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo = typeof(DefaultVillageProductionCalculatorModel).GetMethod("CalculateDailyFoodProductionAmount", BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
```
Comment (tynakuh) [inline src/CommunityPatch/Patches/AgriculturePerkPatch.cs:16]: Should this not be applied on CalculateDailyProductionAmount instead?

File: src/CommunityPatch/Patches/AgriculturePerkPatch.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      });
+    }
+
+    // ReSharper disable once InconsistentNaming
+    private static void Postfix(ref float __result, Village village) {
+      var perk = ActivePatch._perk;
+      if (!(village.Bound?.OwnerClan?.Leader?.GetPerkValue(perk) ?? false))
+        return;
+
+      __result *= perk.PrimaryBonus;
```
Comment (tynakuh) [inline src/CommunityPatch/Patches/AgriculturePerkPatch.cs:57]: I'm assuming result can never be negative here

File: src/CommunityPatch/Patches/AgriculturePerkPatch.cs (hunk 1 above)
Comment (Skau) [reply src/CommunityPatch/Patches/AgriculturePerkPatch.cs:16]: You might be right. We did go back and forth a bit on which of the two to patch.

File: src/CommunityPatch/Patches/AgriculturePerkPatch.cs (hunk 2 above)
Comment (Skau) [reply src/CommunityPatch/Patches/AgriculturePerkPatch.cs:57]: In this case, no. Hearth is never negative.

File: src/CommunityPatch/Patches/AgriculturePerkPatch.cs (hunk 1 above)
Comment (tynakuh) [reply src/CommunityPatch/Patches/AgriculturePerkPatch.cs:16]: > looks like CalculateDailyFoodProductionAmount is actually used to calculate production for food goods in the village in TickFoodProduction in VillageGoodProductionCampaignBehaviour anyways
> so I think CalculateDailyFoodProductionAmount should do the job
> and the result for food production shouldn't be negative, as it just takes the hearthlevel and multiplies it by 0.5 for negative issues

LGTM then

**Pull request merged**